trollsched/shapefiles_from_schedule.py

```diff
@@ -0,0 +1,164 @@
+            try:
+                overpass = create_pass(platform_name, instrument,
+                                       datetime.strptime(child.attrib["start-time"],
+                                                         "%Y-%m-%d-%H:%M:%S"),
+                                       datetime.strptime(child.attrib["end-time"],
+                                                         "%Y-%m-%d-%H:%M:%S"),
+                                       tle_filename=tle_file)
+            except KeyError:
+                continue
+
+            output_filepath = Path(output_dir) / create_shapefile_filename(overpass)
+            create_shapefile_from_pass(overpass, output_filepath)
+
+
+
+def parse_args(args):
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-l", "--log-config",
+                        dest="log_config",
+                        default=None,
+                        required=False,
+                        help="Log config file to use instead of the standard logging.")
+    parser.add_argument("-s", "--satellites",
+                        dest="list_of_satellites",
+                        nargs="*",
+                        default=[],
+                        help="Complete file path to the TLE file to use.",
+                        required=True)
+    parser.add_argument("-t", "--tle_filename",
+                        dest="tle_filepath",
+                        help="Complete file path to the TLE file to use.",
+                        required=True)
+    parser.add_argument("-x", "--xml_filename",
+                        dest="xml_filepath",
+                        help="Complete path to the XML satellite schedule file.",
+                        required=True)
+    parser.add_argument("-o", "--output_dir",
+                        dest="output_dir",
+                        help="Complete path to the XML satellite schedule file.",
+                        default="./")
+    parser.add_argument("-v", "--verbose",
+                        dest="verbosity",
+                        action="count",
+                        default=0,
+                        help="Verbosity (between 1 and 2 occurrences with more leading to more "
+                        "verbose logging). WARN=0, INFO=1, "
+                        "DEBUG=2. This is overridden by the log config file if specified.")
+
+    return parser.parse_args()
+
+
+def run(args=None):
+    """Create shapefiles from an XML schedule of satellite passes."""
+    opts = parse_args(args)
+
+    setup_logging(opts)
+    filename = opts.xml_filepath
+    satellites = opts.list_of_satellites
+    tle_filename = opts.tle_filepath
+    outdir = opts.output_dir
+
+    shapefiles_from_schedule_xml_requests(filename,
+                                          satellites,
+                                          tle_filename,
+                                          outdir)
+
+
+if __name__ == "__main__":
```

Comment:
Should this, the `run()` and `arg_parse()` functions be in a separate "main" script within `bin` directory so they wouldn't affect the coverage? Also the runnable script would be easier to find.

Reply:
Or as a script entrypoint in pyproject.toml, as there doesn't seem to be a `bin/` directory for the executable scripts.

Reply:
This is what is in the `pyproject.toml`:
```
[project.scripts]
schedule = "trollsched.schedule:run"
create_shapefiles_from_schedule = "trollsched.shapefiles_from_schedule:run"
compare_scheds = "trollsched.compare:run"
```

So not sure what you mean?

Reply:
So, I added a basic test of the `shapefiles_from_schedule_xml_requests`. Should we also have a test of the `run` part is that what you mean?
Not sure how I do that properly?